src/Propagation/HTTPTextFormat/HTTPTextFormatInterface.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Propagation\HTTPTextFormat;
+
+use OpenTelemetry\Context\SpanContext;
+use OpenTelemetry\Propagation\Getter\GetterInterface;
+use OpenTelemetry\Propagation\Setter\SetterInterface;
+
+interface HTTPTextFormatInterface
```

Comment:
I don't like this situation:
- We have a triple segmented namespace when I think we should have exactly two segments 1) for the OTEL umbrella and 2) for Trace vs Metrics and so on.
- The name `HTTPTextFormatInterface` is really long, and the `Interface` suffix is unnecessary. Almost entirely the point of OTEL is to have common interfaces for tools to target, so the clean name should be the interface name, not the implementation.

So taking these `use` statements:

```php
use OpenTelemetry\Propagation\Getter\GetterInterface;
use OpenTelemetry\Propagation\Setter\SetterInterface;
```

and, without having read the spec on the subject to propose better names than just `Getter` and `Setter`, I think it should be more like this:

```php
use OpenTelemetry\Propagation\Getter;
use OpenTelemetry\Propagation\Setter;
```

What do the rest of you think?

Reply:
Your implementation idea seems a bit cleaner to me too @morrisonlevi .  A lot more readable.

Reply:
I agree, the `Interface` suffix is unneeded and your suggestion is much easier to read. I was just following how other Interfaces were named in the project.

As for the extra namespaces, I expected there to be quite a few implementations for each of them for all the different carriers that could be used and was concerned the folder might get too crowded but I could be persuaded otherwise.

Reply:
True, it's more readable without the `Interface` suffix. About the carriers - do we really need them if we have the `Setter` interface? One could implement a setter that would modify the carrier to inject the variables. [This](https://github.com/nenad/opencensus-php/blob/master/src/Trace/Propagator/HeaderSetter.php#L11) is how I've dealt with this situation in the OpenCensus implementation.

Reply:
@nenad I believe that's what the spec is meant to do. As you can see in the interface for a `Setter`
```php
public function put($carrier, string $key, string $value) : void;
```
The carrier (Guzzle for example) is passed into `put` and is modified by applying the key/value. So the carriers wouldn't need to be implemented by this library, just the `Getter`s and `Setter`s for them.

After looking over the code again, I would expect there shouldn't be many implementations of `HTTPTextFormat` since the `Getter`s and `Setter`s will be so flexible so I wouldn't be opposed to moving that back to the `\OpenTelemetry\Propogation` namespace.